diff --git a/toolchains/cypress/BUILD.bazel b/toolchains/cypress/BUILD.bazel
index b112c168a9..1d59a2f043 100644
--- a/toolchains/cypress/BUILD.bazel
+++ b/toolchains/cypress/BUILD.bazel
@@ -28,6 +28,7 @@ toolchain_type(name = "toolchain_type")
     cypress_files = "@cypress_%s//:files" % os_name,
 ) for os_name in [
     "darwin",
+    "darwin_arm64",
     "linux",
     "windows",
 ]]
@@ -37,7 +38,8 @@ toolchain_type(name = "toolchain_type")
 alias(
     name = "toolchain",
     actual = select({
-        "@bazel_tools//src/conditions:darwin": ":cypress_darwin_toolchain_config",
+        "@bazel_tools//src/conditions:darwin_x86_64": ":cypress_darwin_toolchain_config",
+        "@bazel_tools//src/conditions:darwin_arm64": ":cypress_darwin_arm64_toolchain_config",
         "@bazel_tools//src/conditions:linux_x86_64": ":cypress_linux_toolchain_config",
         "@bazel_tools//src/conditions:windows": ":cypress_windows_toolchain_config",
         "//conditions:default": ":cypress_linux_toolchain_config",
@@ -49,7 +51,8 @@ alias(
 alias(
     name = "cypress_bin",
     actual = select({
-        "@bazel_tools//src/conditions:darwin": "@cypress_darwin//:bin",
+        "@bazel_tools//src/conditions:darwin_x86_64": "@cypress_darwin//:bin",
+        "@bazel_tools//src/conditions:darwin_arm64": "@cypress_darwin_arm64//:bin",
         "@bazel_tools//src/conditions:linux_x86_64": "@cypress_linux//:bin",
         "@bazel_tools//src/conditions:windows": "@cypress_windows//:bin",
         "//conditions:default": "@cypress_linux//:bin",
@@ -77,6 +80,16 @@ toolchain(
     toolchain_type = ":toolchain_type",
 )
 
+toolchain(
+    name = "cypress_darwin_arm64_toolchain",
+    target_compatible_with = [
+        "@platforms//os:osx",
+        "@platforms//cpu:arm64",
+    ],
+    toolchain = ":cypress_darwin_arm64_toolchain_config",
+    toolchain_type = ":toolchain_type",
+)
+
 toolchain(
     name = "cypress_windows_toolchain",
     target_compatible_with = [
diff --git a/toolchains/cypress/cypress_repositories.bzl b/toolchains/cypress/cypress_repositories.bzl
index a55f43f40d..a0a4ef9444 100644
--- a/toolchains/cypress/cypress_repositories.bzl
+++ b/toolchains/cypress/cypress_repositories.bzl
@@ -24,6 +24,8 @@ def cypress_repositories(
         linux_sha256 = "",
         darwin_urls = [],
         darwin_sha256 = "",
+        darwin_arm64_urls = [],
+        darwin_arm64_sha256 = "",
         windows_urls = [],
         windows_sha256 = ""):
     """
@@ -34,8 +36,10 @@ def cypress_repositories(
         version: Version of cypress binary to use. Should match package.json
         linux_urls: (Optional) URLs at which the cypress binary for linux distros of linux can be downloaded. If omitted, https://cdn.cypress.io/desktop will be used.
         linux_sha256: (Optional) SHA-256 of the linux cypress binary
-        darwin_urls: (Optional) URLs at which the cypress binary for darwin distros of linux can be downloaded. If omitted, https://cdn.cypress.io/desktop will be used.
+        darwin_urls: (Optional) URLs at which the cypress binary for darwin can be downloaded. If omitted, https://cdn.cypress.io/desktop will be used.
         darwin_sha256: (Optional) SHA-256 of the darwin cypress binary
+        darwin_arm64_urls: (Optional) URLs at which the cypress binary for darwin arm64 can be downloaded. If omitted, https://cdn.cypress.io/desktop will be used (note: as of this writing (11/2021), Cypress does not have native arm64 builds, and this URL will link to the x86_64 build to run under Rosetta).
+        darwin_arm64_sha256: (Optional) SHA-256 of the darwin arm64 cypress binary
         windows_urls: (Optional) URLs at which the cypress binary for windows distros of linux can be downloaded. If omitted, https://cdn.cypress.io/desktop will be used.
         windows_sha256: (Optional) SHA-256 of the windows cypress binary
     """
@@ -74,6 +78,32 @@ filegroup(
     visibility = ["//visibility:public"],
 )
 
+filegroup(
+    name = "bin",
+    # Cypress checks that the binary path matches **/Contents/MacOS/Cypress
+    srcs = ["Cypress.app/Contents/MacOS/Cypress"],
+    visibility = ["//visibility:public"],
+)
+""",
+    )
+
+    http_archive(
+        name = "cypress_darwin_arm64".format(name),
+        sha256 = darwin_arm64_sha256,
+        # Cypress checks that the binary path matches **/Contents/MacOS/Cypress so we do not strip that particular prefix.
+        urls = darwin_arm64_urls + [
+            # Note: there is currently no arm64 builds of cypress, so here we'll default to
+            # the x64 version so apple silicon macs can run the binary using Rosetta.
+            # Once a native arm64 build is available, this should be updated
+            "https://cdn.cypress.io/desktop/{}/darwin-x64/cypress.zip".format(version),
+        ],
+        build_file_content = """
+filegroup(
+    name = "files",
+    srcs = ["Cypress.app"],
+    visibility = ["//visibility:public"],
+)
+
 filegroup(
     name = "bin",
     # Cypress checks that the binary path matches **/Contents/MacOS/Cypress
@@ -105,6 +135,6 @@ filegroup(
     )
 
     # This needs to be setup so toolchains can access nodejs for all different versions
-    for os_name in ["windows", "darwin", "linux"]:
+    for os_name in ["windows", "darwin", "darwin_arm64", "linux"]:
         toolchain_label = Label("@build_bazel_rules_nodejs//toolchains/cypress:cypress_{}_toolchain".format(os_name))
         native.register_toolchains("@{}//{}:{}".format(toolchain_label.workspace_name, toolchain_label.package, toolchain_label.name))